java/ql/src/Security/CWE/CWE-190/BuggyDivideAndConquer.ql

```diff
@@ -0,0 +1,46 @@
+/**
+ * @name Query for detecting potential overflow in divide and conquer algorithms leading to crash
+ * @description Divide and conquer algorithms (binary search, merge sort, ...)
+ *              will fail for large values, when calculating the mid-point overflows.
+ *              This can lead to an ArrayIndexOutOfBoundsException, if the (negative!) mid-point is used as an array index. 
+ * @kind problem
+ * @problem.severity error
```

Comment:
"error" seems too strong, there are likely many cases where the added ints are known to not overflow.